src/compute/src/sink/materialized_view_v2.rs

```diff
@@ -1075,6 +1122,7 @@ mod append {
                     batch.lower().elements(),
                     batch.upper().elements(),
                 ));
+                batch.delete().await;
```

Comment:
Maybe add a comment here why `batch.delete()` is necessary?

Reply:
Added one! I didn't put in the detail about this not actually doing anything right now because persist just leaks the batch data anyway, because persist might change and I don't want that comment to become stale when it does.